src/main/java/org/owasp/wrongsecrets/challenges/docker/binaryexecution/BinaryExecutionHelper.java

```diff
@@ -73,31 +73,38 @@ public String executeCommand(String guess, String fileName) {
 
     @SuppressFBWarnings(value = "COMMAND_INJECTION", justification = "We check for various injection methods and counter those")
     private String executeCommand(File execFile, Operation operation, String guess) throws IOException, InterruptedException {
-        ProcessBuilder ps;
 
-        if (!execFile.getPath().contains("wrongsecrets") || stringContainsCommandChainToken(execFile.getPath())
-            || stringContainsCommandChainToken(guess)) {
+
+        if (isInvalidInput(execFile, guess)) {
             return BinaryExecutionHelper.ERROR_EXECUTION;
         }
+
+        CommandExecutor commandExecutor;
```

Comment:
extracting methods to boolean more descriptive methods is fine. This just makes it harder to maintain the code: for me there are now more class files we have to take care off. It would be very sensible if there are more commands to be executed in the future, while the opposite will be true: i will most likely redesign the binaries to actually always take the same set of parameters. (so, can you please undo this :) ?)